sympy/functions/elementary/trigonometric.py

```diff
@@ -2222,7 +2222,7 @@ class asec(InverseTrigonometricFunction):
     @classmethod
     def eval(cls, arg):
         if arg.is_Number:
-            if arg is S.NaN:
+            if arg in [S.NaN, S.Zero]:
```

Comment:
I don't think so.
in `eval` method
adding an if statement like

```
if arg.is_zero:
  return S.NaN
```
in the starting should do.